lib/autocomplete.ts

```diff
@@ -61,9 +53,26 @@ export function autocomplete<T>(
     ...options
   };
 
-  const contentFunc = options.contentFunc!;
+  // Keeps track of the last value as typed by the user.
+  let lastAsTyped = inputElem.value;
+  const lis = dom.onElem(inputElem, 'input', () => lastAsTyped = inputElem.value);
+
+  const contentFunc = () => [
+    dom.autoDispose(lis),
+    dom.forEach(choices, (opt) => (
+      menuItem(
+        () => { inputElem.value = opt; if (options.onClick) { options.onClick(opt); }},
+        opt,
+        dom.data('choice', opt),
+        options.updateOnSelect ?
+          dom.data('menuItemSelected', () => (yesNo: boolean) => inputElem.value = (yesNo ? opt : lastAsTyped)) :
```

Comment:
I can't understand what this does. Every time a menu item is selected, it constructs a new function. Does it ever call this function? Is something broken in grainjs? I'd expect that after `dom.data('foo', bar)`, `dom.getData(elem, 'foo')` would return `bar`; and the code in menu simply calls it when its set. Something somewhere is definitely not what it should be.

Reply:
That code is removed now. 
Still `dom.data` is a little trickier than that. It expects the 2nd argument to be a bindable value, so it you pass in a function it will create a computed out of it.